src/parameters/update_cost_parameters.jl

```diff
@@ -14,31 +14,62 @@ function _update_parameter_values!(
     template = get_template(model)
     device_model = get_model(template, V)
     components = get_available_components(device_model, get_system(model))
-
     for component in components
         if _has_variable_cost_parameter(component)
             name = PSY.get_name(component)
-            ts_vector = PSY.get_variable_cost(
-                component,
-                PSY.get_operation_cost(component);
-                start_time = initial_forecast_time,
-                len = horizon,
-            )
-            variable_cost_forecast_values = TimeSeries.values(ts_vector)
-            for (t, value) in enumerate(variable_cost_forecast_values)
-                if attributes.uses_compact_power
-                    # TODO implement this
-                    value, _ = _convert_variable_cost(value)
-                end
-                # TODO removed an apparently unused block of code here?
-                _set_param_value!(parameter_array, value, name, t)
-                update_variable_cost!(
-                    container,
-                    parameter_array,
-                    parameter_multiplier,
-                    attributes,
+            op_cost = PSY.get_operation_cost(component)
+            if op_cost isa PSY.MarketBidCost
+                ts_vector = PSY.get_variable_cost(
                     component,
-                    t,
+                    PSY.get_operation_cost(component);
+                    start_time = initial_forecast_time,
+                    len = horizon,
+                )
+                variable_cost_forecast_values = TimeSeries.values(ts_vector)
+                for (t, value) in enumerate(variable_cost_forecast_values)
+                    if attributes.uses_compact_power
+                        # TODO implement this
+                        value, _ = _convert_variable_cost(value)
+                    end
+                    # TODO removed an apparently unused block of code here?
+                    _set_param_value!(parameter_array, value, name, t)
+                    update_variable_cost!(
+                        container,
+                        parameter_array,
+                        parameter_multiplier,
+                        attributes,
+                        component,
+                        t,
+                    )
+                end
+            elseif op_cost isa PSY.ThermalGenerationCost
+                fuel_curve = PSY.get_variable(op_cost)
+                ts_vector = PSY.get_fuel_cost(
+                    component;
+                    start_time = initial_forecast_time,
+                    len = horizon,
+                )
+                fuel_cost_forecast_values = TimeSeries.values(ts_vector)
+                for (t, value) in enumerate(fuel_cost_forecast_values)
+                    if attributes.uses_compact_power
+                        # TODO implement this
```

Comment:
same here

Reply:
These TODOs were there before this PR. Not sure if we can safely remove those TODO.